howtos/diffs/distributed-training.diff

```diff
@@ -1,26 +1,33 @@
-diff --git a/examples/mnist/train.py b/examples/mnist/train.py
-index 9c21985..fed5421 100644
---- a/examples/mnist/train.py
-+++ b/examples/mnist/train.py
-@@ -23,6 +23,9 @@ from absl import app
- from absl import flags
+diff --git a/examples/mnist/mnist_lib.py b/examples/mnist/mnist_lib.py
+index 63af6e8..87c57b0 100644
+--- a/examples/mnist/mnist_lib.py
++++ b/examples/mnist/mnist_lib.py
+@@ -20,6 +20,8 @@ The data is loaded using tensorflow_datasets.
+ 
  from absl import logging
  
 +import functools
 +
+ import jax
+ from jax import random
+ 
+@@ -29,6 +31,7 @@ import numpy as onp
+ 
+ import tensorflow_datasets as tfds
+ 
 +from flax import jax_utils
  from flax import nn
  from flax import optim
  from flax.metrics import tensorboard
-@@ -87,6 +90,7 @@ def create_model(key):
+@@ -60,6 +63,7 @@ def create_model(key):
  def create_optimizer(model, learning_rate, beta):
    optimizer_def = optim.Momentum(learning_rate=learning_rate, beta=beta)
    optimizer = optimizer_def.create(model)
 +  optimizer = jax_utils.replicate(optimizer)
    return optimizer
  
  
-@@ -99,6 +103,11 @@ def cross_entropy_loss(logits, labels):
+@@ -72,6 +76,11 @@ def cross_entropy_loss(logits, labels):
```

Comment:
I'm curious about these -- looks like when base examples change, many of the HOWTO diff changes lead to these essentially informationless changes, just because line numbers change.

Does `git apply` even look at line numbers or just the context around them? Could we perhaps somehow not include line numbers in our diffs?